### PR TITLE
MSA SAXS model

### PR DESCRIPTION
**Pull request by stephprince**
## Motivation

This is a PR of the progress I have so far for the MSA SAXS model. This code uses OpenFold modules or redefines them where necessary to incorporate the SAXS data (similar to AlphaFlow's setup). 

TODO list:
- [X] check that a training loop successfully runs
- [x] check that logging and checkpoints work as expected
- [x] optimize training parameters for performance
- [x] run on subset of pdb70 training dataset and monitor results
- [x] run on full pdb70 training dataset with frozen af weights
- [x] run on full pdb70 training dataset with unfrozen af weights  
- [x] add methods and scripts for inference 
- [x] evaluate results on apo/holo data
- [x] evaluate results on enm/nma data
- [x] add cross-attention to pair distribution in addition to msa
- [x] add saxs component to loss function
- [x] generate synthetic dataset for training
- [x] evaluate different weights for saxs loss component
- [x] rerun training with new model and synthetic dataset
- [x] decide training stopping point
- [x] inspect model outputs, develop evaluation analyses

## How to test the behavior?
```
```

## Checklist

- [ ] Did you update CHANGELOG.md with your changes?
- [ ] Have you checked our [Contributing](https://github.com/exabiome/gtnet/blob/dev/docs/CONTRIBUTING.rst) document?
- [ ] Have you ensured the PR clearly describes the problem and the solution?
- [ ] Is your contribution compliant with our coding style? This can be checked running `ruff` from the source directory.
- [ ] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/exabiome/gtnet/pulls) for the same change?
- [ ] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: src/metfish/msa_model/model/alphafold_saxs.py [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+    atom14_to_atom37,
+)
+from openfold.utils.tensor_utils import (
+    add,
+    tensor_tree_map,
+)
+
+
+# define the model
+class SAXSMSAAttention(nn.Module):
```
Comment (stephprince) [inline src/metfish/msa_model/model/alphafold_saxs.py:53]: @ajtritt @smallfishabc could you take a look at this SAXSMSAAttention module whenever you get a chance to let me know if it looks like a reasonable place to start?

Comment (smallfishabc) [reply src/metfish/msa_model/model/alphafold_saxs.py:53]: I think it is reasonable. I only have one question about the q_x. How did you make the second dimension to one?

**Review by smallfishabc — DISMISSED** [review]

**Review by ajtritt — APPROVED** [review]
